lib/seattleflu/id3c/cli/command/etl/redcap_det_uw_reopening.py

```diff
@@ -0,0 +1,1312 @@
+            follow_up_encounter_entry, follow_up_encounter_reference = create_follow_up_encounter(
+                redcap_record_instance, patient_reference, site_reference, initial_encounter_reference)
+            follow_up_questionnaire_entry = create_follow_up_questionnaire_response(
+            redcap_record_instance, patient_reference, follow_up_encounter_reference)
+            follow_up_computed_questionnaire_entry = create_computed_questionnaire_response(
+            redcap_record_instance, patient_reference, follow_up_encounter_reference,
+            birthdate, datetime.strptime(follow_up_encounter_entry['resource']['period']['start'], '%Y-%m-%d'))
+
+        current_entries = [
+            initial_encounter_entry,
+            computed_questionnaire_entry,
+            enrollment_questionnaire_entry,
+            testing_determination_internal_questionnaire_entry,
+            daily_questionnaire_entry,
+            specimen_entry,
+            specimen_observation_entry,
+            follow_up_encounter_entry,
+            follow_up_questionnaire_entry,
+            follow_up_computed_questionnaire_entry
+        ]
+
+        resource_entries.extend(list(filter(None, current_entries)))
+
+
+    return create_bundle_resource(
+        bundle_id = str(uuid4()),
+        timestamp = datetime.now().astimezone().isoformat(),
+        source = f"{REDCAP_URL}{project_id}/{record_id}",
+        entries = list(filter(None, resource_entries))
+    )
+
+
+def create_site_reference(record: dict, collection_method: CollectionMethod, event_type: EventType) -> Optional[Dict[str,dict]]:
+    """
+    Create a Location reference for site of the sample collection encounter based
+    on how the sample was collected.
+    """
+    if collection_method == CollectionMethod.KIOSK:
+        record_location = record.get('location_type')
+        if record_location:
+            site = site_map(record_location)
+    else:
+        site = 'uwReopeningSwabNSend'
+
+    return {
+        "location": create_reference(
+            reference_type = "Location",
+            identifier = create_identifier(f"{INTERNAL_SYSTEM}/site", site)
+        )
+    }
+
+
+def site_map(record_location: str) -> str:
+    """
+    Maps *record_location* to the corresponding site name.
+    """
+    location_site_map = {
+        'bothell':  'UWBothell',
+        'odegaard': 'UWOdegaardLibrary',
+        'slu':      'UWSouthLakeUnion',
+        'tacoma':   'UWTacoma',
+        'uw_club':  'UWClub'
+    }
+
+    if record_location not in location_site_map:
+        raise UnknownRedcapRecordLocation(f"Found unknown location type «{record_location}»")
+
+    return location_site_map[record_location]
+
+
+def locations(db: DatabaseSession, cache: TTLCache, record: dict) -> list:
+    """ Creates a list of Location resource entries from a REDCap record. """
+    lodging_options = [
```

Comment:
Adding dorm into this location list could be a nice-to-have

Reply:
@seattleflu/id3c-devs , would dorm name fit into locations? Right now we're using housing_type (residence or lodging) and the physical address.

Reply:
That's a good question. I remember on a recent call that we were saying dorm probably counts as PII, so it might not be a good idea to ingest it directly. Perhaps I'm not remembering correctly, though.

Reply:
On a call last week we talked about not ingesting dorm_room (the actual dorm room number). I am currently ingesting the dorm name, the greek house name, and the UW apartment building name. It would be handy to do analysis of that data in ID3C (for surges, for example).  But if we don't ingest things like zipcode out of PII concern, it would seem that I should not bring in data that would identify an actual building. @kairstenfay @joverlee521 , do you agree with me *not* bringing in building names like those above?

We could still use the dorm to create a location even if we didn't store the dorm name itself, so that's still an open question.

Reply:
Here's what the group discussion in Slack came up with:
Do not ingest building names: Greek house, dorm, UW apartment name. Those are PII.
For creating locations:
1. Create the location for home address provided.
2. Look up a geocode for the Greek house, dorm, or UW apartment name. If we get one and if it's not the same as their home address, add it as a second location.

Reply:
I created Trello card https://trello.com/c/RWeUF0DA/814-in-uw-reopening-etl-add-a-location-for-dorm-greek-house-or-uw-apartment for this. This won't be in for go-live but could be added later.